components/sup/src/cli.rs

```diff
@@ -44,6 +105,224 @@ mod test {
+
+        assert_cli_cmd!(ring_should_have_short_flag,
+                        "hab-sup run -r flippitygibbit",
+                        "RING" => "flippitygibbit");
+
+        #[test]
+        fn ring_conflicts_with_ring_key() {
+            let cmd_vec =
+                Vec::from_iter("hab-sup run --ring myring --ring-key foobar".split_whitespace());
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        assert_cli_cmd!(should_handle_ring_key,
+                        "hab-sup run --ring-key secrit",
+                        "RING_KEY" => "secrit");
+
+        assert_cli_cmd!(should_handle_channel,
+                        "hab-sup run --channel dev",
+                        "CHANNEL" => "dev");
+
+        assert_cli_cmd!(should_handle_bldr_url,
+                        "hab-sup run --url http://not.a.real.com",
+                        "BLDR_URL" => "http://not.a.real.com");
+
+        assert_cli_cmd!(bldr_url_should_have_a_short_flag,
+                        "hab-sup run -u http://not.a.real.com",
+                        "BLDR_URL" => "http://not.a.real.com");
+
+        assert_cli_cmd!(bldr_url_should_have_a_default_value,
+                        "hab-sup run",
+                        "BLDR_URL" => "https://bldr.habitat.sh");
+
+        #[test]
+        fn invalid_bldr_url_should_fail() {
+            let cmd_vec = Vec::from_iter("hab-sup run -u bad-url".split_whitespace());
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        #[test]
+        fn should_handle_valid_config_dir() {
+            let config_dir = TempDir::new().expect("Could not create tempdir");
+            let path_str = config_dir.path().to_str().unwrap();
+            let cmd_str = format!("hab-sup run --config-from {}", path_str);
+            let cmd_vec = Vec::from_iter(cmd_str.split_whitespace());
+            let matches = cli()
+                .get_matches_from_safe(cmd_vec)
+                .expect("Could not get ArgMatches");
+            let subcommand = matches
+                .subcommand_matches("run")
+                .expect("Failed to get matches for sup run");
+            assert_eq!(subcommand.value_of("CONFIG_DIR"), Some(path_str));
+        }
+
+        #[test]
+        fn config_dir_should_fail_when_dir_does_not_exist() {
+            let cmd_vec = Vec::from_iter(
+                "hab-sup run --config-from really-fake/not-real/directory/no-really/go-away"
+                    .split_whitespace(),
+            );
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        assert_cli_cmd!(should_handle_auto_update,
+                        "hab-sup run --auto-update",
+                        "AUTO_UPDATE" => true);
+
+        assert_cli_cmd!(auto_update_should_have_a_short_flag,
+                        "hab-sup run -A",
+                        "AUTO_UPDATE" => true);
+
+        #[test]
+        fn should_take_cert_file_and_key_file() {
```

Comment:
This seems like too much logic to be worthwhile. The clap declarations `{file_exists} requires[CERT_FILE]` and `{file_exists} requires[KEY_FILE]` are pretty straightforward. Similarly for the next two tests.

Reply:
Yeah it is a good bit of logic, but I want to make sure that I've captured everything. I don't want to refactor `hab_sup_run` and accidentally leave something out.